packages/yourloops/components/device/parameter-change-value.tsx

```diff
@@ -25,40 +25,38 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import styles from '../diabeloop.css'
-import React, { type FunctionComponent } from 'react'
-import { type HistorizedParameter } from '../../../models/historized-parameter.model'
-import type moment from 'moment-timezone'
+import React, { type FC } from 'react'
+import { type ParametersChange, type PumpSettingsParameter } from 'medical-domain'
+import Box from '@mui/material/Box'
+import ChevronRightIcon from '@mui/icons-material/ChevronRight'
+import { useTheme } from '@mui/material/styles'
 
-interface HistorySpannedRowProps {
-  data: HistorizedParameter
-  onSwitchToDaily: (date: moment.Moment | Date | number | null) => void
+interface ParameterChangeValueProps {
+  historyCurrentIndex: number
+  history: ParametersChange[]
+  parameter: PumpSettingsParameter
 }
 
-const ONE_SPACE_STRING = '&nbsp;'
-const NUMBER_OF_COLUMNS = 4
+export const ParameterChangeValue: FC<ParameterChangeValueProps> = (props) => {
+  let previousParameter: PumpSettingsParameter
+  const theme = useTheme()
+  const { parameter, history, historyCurrentIndex } = props
+  const filteredHistory = history.slice(historyCurrentIndex + 1)
 
-export const HistorySpannedRow: FunctionComponent<HistorySpannedRowProps> = (props) => {
-  const { onSwitchToDaily, data } = props
-  const content = data.groupedParameterHeaderContent ?? ONE_SPACE_STRING
+  filteredHistory.every(parametersChange => {
```

Comment:
What's the point of using `every` for this? You don't use the returned value so it should be a `foreach` :thinking: 

Can you explain to me what this does?

Reply:
I need to stop the loop as soon as the previous value has been found. The every method can do that if we return a false value. If I use a forEach, previous parameter will be overwritten if another is found later in the array.